src/apps/intermediate-list/intermediate-list.tsx

```diff
@@ -0,0 +1,112 @@
+import { useText } from "../../core/utils/text";
+import { useUrls } from "../../core/utils/url";
+import FeeList from "./fee-list/fee-list";
+import FeeDetailsModal from "./modal/fee-details-modal";
+import MyPaymentOverviewModal from "./modal/my-payment-overview-modal";
+import FeeDetailsContent from "./stackable-fees/fee-details-content";
+import {
+  getFeeObjectByFaustId,
+  getFeesPostPaymentChangeDate,
+  getFeesPrePaymentChangeDate
+} from "./utils/intermediate-list-helper";
+
+const IntermedateList: FC = () => {
+  const t = useText();
+  const { open } = useModalButtonHandler();
+  const { modalIds } = useSelector((s: ModalIdsProps) => s.modal);
+  const { data: fbsFees = [] } = useGetFeesV2<FeeV2[]>();
+  const [itemsPrePaymentChange, setItemsPrePaymentChange] = useState<
+    FeeV2[] | null
+  >(null);
+  const [totalFeePrePaymentChange, setTotalFeePrePaymentChange] =
+    useState<number>(0);
+  const [itemsPostPaymentChange, setItemsPostPaymentChange] = useState<
+    FeeV2[] | null
+  >(null);
+  const [totalFeePostPaymentChange, setTotalFeePostPaymentChange] =
+    useState<number>(0);
+  const { viewFeesAndCompensationRatesUrl } = useUrls();
+  const [faustIdentifier, setFaustIdentifier] = useState("0");
+  const [feeDetailsData, setFeeDetailsData] = useState<FeeV2[]>();
+  const openDetailsModalClickEvent = useCallback(
+    (faustId: string) => {
+      setFaustIdentifier(faustId);
+      setFeeDetailsData(getFeeObjectByFaustId(fbsFees, faustId));
+      open(faustId || "");
+    },
+    [fbsFees, open]
+  );
+
+  useEffect(() => {
+    if (fbsFees) {
+      const feesPrePaymentChange = getFeesPrePaymentChangeDate(fbsFees).length;
+      if (feesPrePaymentChange > 0) {
+        setItemsPrePaymentChange(getFeesPrePaymentChangeDate(fbsFees));
+      }
+      const feesPostPaymentChange = getFeesPrePaymentChangeDate(fbsFees).length;
+      if (feesPostPaymentChange > 0) {
+        setItemsPostPaymentChange(getFeesPostPaymentChangeDate(fbsFees));
+      }
+    }
+  }, [fbsFees]);
+
+  useEffect(() => {
+    let totalFee = 0;
+    itemsPrePaymentChange?.forEach((item) => {
+      totalFee += item.amount;
+    });
+    setTotalFeePrePaymentChange(totalFee);
+  }, [itemsPrePaymentChange]);
+
+  useEffect(() => {
+    let totalTally = 0;
+    itemsPostPaymentChange?.forEach((item) => {
+      totalTally += item.amount;
+    });
+    setTotalFeePostPaymentChange(totalTally);
+  }, [itemsPostPaymentChange]);
+  return (
+    <>
+      <div
+        style={modalIds.length > 0 ? { display: "none" } : {}}
+        className="intermediate-list-page"
+      >
+        <h1 className="intermediate-list-headline">
+          {t("intermediateListHeadlineText")}
+        </h1>
+        <span className="intermediate-list-body">
+          {t("intermediateListBodyText")}{" "}
+          <Link
+            className="link-tag"
+            href={new URL(viewFeesAndCompensationRatesUrl)}
```

Comment:
It should not be necessary to create an URL object here. It is a part of the functionality of the useUrls()hook to transform the strings to URL objects.